---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Cmd+Shift+Enter*. 
<!-- Ralimetinib (LY2228820)  -->
# Ralimetinib
# (LY2228820)
```{r}
DOI_broad_id=onTarget$drugCategory$broad_id_trimmed[grep('LY2228820', onTarget$drugCategory$name)]
onTarget$drugCategory[grep('LY2228820', onTarget$drugCategory$name),]
GOI='MAPK14'
```
<!-- Extract response -->
```{r}
drug_resp=onTarget$drug_prism[DOI_broad_id,]
```
<!-- Compute correlation -->
```{r}
matched_cellLines=intersect(colnames(onTarget$avana), names(drug_resp))
correlation_withdrug=apply(onTarget$avana[,matched_cellLines], 1, function(x)
  unlist(cor.test_trimmed_v0(x, drug_resp[matched_cellLines]))
  )
```
<!-- Compute correlation -->
```{r}
correlation_withdrug_df=as.data.frame(t(correlation_withdrug))
correlation_withdrug_df=correlation_withdrug_df[order(correlation_withdrug_df$estimate.cor, decreasing = T),]
correlation_withdrug_df$geneRank=1:nrow(correlation_withdrug_df)
correlation_withdrug_df['EGFR',]; correlation_withdrug_df['MAPK14',]
```
<!-- Secondary Targets -->
<!-- Step 0: Find cell lines with low expression -->
```{r}
LowprimaryTargets=names(which(vectorSubset(onTarget$expression_20Q4[GOI,],matched_cellLines)>5))
```

```{r}
compute_cor_for_a_drug<- function(infunc_drugID=DOI_broad_id,
                                  infunc_COI=LowprimaryTargets){
  # KO data
  infunc_drugResp=drug_resp[LowprimaryTargets]
  # COI stands for Cell lines of Interest
  # Cmpute correlation
  correlation_mat=apply(onTarget$avana[,infunc_COI], 1, function(x)
    unlist(cor.test_trimmed_v0(x, infunc_drugResp))
  )
  correlation_mat=as.data.frame(t(correlation_mat))
  correlation_mat=correlation_mat[
    order(correlation_mat$estimate.cor, decreasing = T),]
  correlation_mat$geneRank=1:nrow(correlation_mat)
  correlation_mat
}
correlation_mat[GOI,]
```
<!-- Mutant vs WT correlation to primary target -->
```{r}
PrimaryTarget='EGFR'
PrimaryTarget_mutation=onTarget$mutations_matrix[PrimaryTarget,]
mutant_cellLines=names(which(PrimaryTarget_mutation==1))
WT_cellLines=names(which(PrimaryTarget_mutation==0))
# Correlation in WT vs Mutant
compute_cor_for_a_drugMutantvsWT<- function(infunc_drugID=DOI_broad_id,
                                            infunc_COI=mutant_cellLines){
  # KO data
  infunc_COI=intersect(infunc_COI, matched_cellLines)
  infunc_drugResp=drug_resp[infunc_COI]
  # COI stands for Cell lines of Interest
  # Cmpute correlation
  correlation_mat=apply(onTarget$avana[,infunc_COI], 1, function(x)
    unlist(cor.test_trimmed_v0(x, infunc_drugResp))
  )
  correlation_mat=as.data.frame(t(correlation_mat))
  correlation_mat=correlation_mat[
    order(correlation_mat$estimate.cor, decreasing = T),]
  correlation_mat$geneRank=1:nrow(correlation_mat)
  correlation_mat
}
inMutant=compute_cor_for_a_drugMutantvsWT(infunc_drugID=DOI_broad_id,infunc_COI=mutant_cellLines)
inWT=compute_cor_for_a_drugMutantvsWT(infunc_drugID=DOI_broad_id,infunc_COI=WT_cellLines)
inMutant['EGFR',]
inWT['EGFR',]
```
<!-- Load CCLE mutation data-->
```{r}
CCLE_mutations=read.csv('/Users/sinhas8/Project_TrueTarget/Data/CCLE_mutations.csv')
where_crispr_drug_also_available=CCLE_mutations$DepMap_ID %in% matched_cellLines
CCLE_mutations_matched=CCLE_mutations[where_crispr_drug_also_available,]
# remove silent
CCLE_mutations_matched=CCLE_mutations_matched[CCLE_mutations_matched$Variant_Classification!='Silent',]
# <!-- Find the most frequent variant -->
egfr_id=CCLE_mutations_matched$Hugo_Symbol %in% 'EGFR'
sort(table(CCLE_mutations_matched$Protein_Change[egfr_id]), decreasing = T)
sort(table(CCLE_mutations_matched$cDNA_Change[egfr_id]), decreasing = T)
sort(table(CCLE_mutations_matched$Genome_Change[egfr_id]), decreasing = T)

CCLE_mutations_matched[egfr_id & as.logical(CCLE_mutations_matched$isCOSMIChotspot),]
# "Protein_Change" columnd has variant informations
CCLE_mutations_matched$gene_variant=paste(CCLE_mutations_matched$Protein_Change, CCLE_mutations_matched$Hugo_Symbol, sep=';')
genes_wd_drugs_available=CCLE_mutations_matched$Hugo_Symbol %in% KnownTarget_predictions$MaxTargetName
Protein_Change_freq=table(CCLE_mutations_matched$gene_variant[genes_wd_drugs_available])
Protein_Change_freq_filteresLowout=Protein_Change_freq[Protein_Change_freq>5]
Protein_Change_freq_filteresLowout=sort(Protein_Change_freq_filteresLowout, decreasing = T)
most_frequent_varints=do.call(rbind, strsplit(names(Protein_Change_freq_filteresLowout), split = ';'))
most_frequent_varints_df=data.frame(most_frequent_varints)
colnames(most_frequent_varints_df)=c('variant', 'gene')
most_frequent_varints_df$variant_clean=gsub('p.','',most_frequent_varints_df$variant)
most_frequent_varints_df$freq=Protein_Change_freq_filteresLowout
```

<!-- Repeat correlation in a given cancer type -->
```{r}
compute_cor_for_a_drug<- function(infunc_drugName='TG-02',
                                  COI=matched_cellLines){
  # KO data
  tg02_id=grep(infunc_drugName,onTarget$drugsCommonName)
  broad_id_tg02=onTarget$drugCategory$broad_id_trimmed[
    grep('TG-02',onTarget$drugCategory$name)]
  tg02_crispr=onTarget$secondary_prism[broad_id_tg02,]
  # COI stands for Cell lines of Interest
  # Cmpute correlation
  COI_matched=intersect(COI, matched_cellLines)
  correlation_mat=apply(onTarget$avana[,COI_matched], 1, function(x)
    unlist(cor.test_trimmed_v0(x, tg02_crispr[COI_matched]))
  )
  correlation_mat=as.data.frame(t(correlation_mat))
  correlation_mat=correlation_mat[
    order(correlation_mat$estimate.cor, decreasing = T),]
  correlation_mat$geneRank=1:nrow(correlation_mat)
  correlation_mat
}
```
<!-- GBM cell lines -->
<!-- We do not have enough GBM celllines -->
```{r}
gbm_cellLines=onTarget$annotation$depMapID[which(onTarget$annotation$tcga_code=='GBM')]
correlation_withdrug_gbm=compute_cor_for_a_drug(infunc_drugName='TG-02',COI=gbm_cellLines)
correlation_withdrug_gbm[grep('CDK9',rownames(correlation_withdrug_gbm)),]
correlation_withdrug_gbm[grep('LTA',rownames(correlation_withdrug_gbm)),]
```

<!-- IDH mutant cell lines -->
```{r}
idh_id=which(rownames(onTarget$mutations_matrix)=='IDH1' | 
               rownames(onTarget$mutations_matrix)=='IDH2')
idh_mutation_status=apply(onTarget$mutations_matrix[idh_id,], 2, max)
IDHMutant_cellLines=which(as.logical(idh_mutation_status))
IDHMutant_cellLines=names(idh_mutation_status)[IDHMutant_cellLines]
correlation_withdrug_idh=compute_cor_for_a_drug(infunc_drugName='TG-02',
                                                COI=IDHMutant_cellLines)
correlation_withdrug_idh[grep('CDK9',rownames(correlation_withdrug_idh)),]
correlation_withdrug_idh[grep('LTA',rownames(correlation_withdrug_idh)),]
```


Add a new chunk by clicking the *Insert Chunk* button on the toolbar or by pressing *Cmd+Option+I*.
When you save the notebook, an HTML file containing the code and output will be saved alongside it (click the *Preview* button or press *Cmd+Shift+K* to preview the HTML file). 

The preview shows you a rendered HTML copy of the contents of the editor. Consequently, unlike *Knit*, *Preview* does not run any R code chunks. Instead, the output of the chunk when it was last run in the editor is displayed.

